src/components/DownloadImageDialog/ImageForm.tsx

```diff
@@ -124,29 +121,41 @@ export const ImageForm = memo(function ImageForm({
 	);
 
 	const handleVariantChange = useCallback(
-		(v: typeof variant) => {
-			setVariant(v);
-			onChange('developmentMode', v === 'dev');
+		(newVariant: typeof variant) => {
+			setVariant(newVariant);
+			const newState: Partial<DownloadImageFormModel> = {
+				developmentMode: newVariant === 'dev',
+			};
+			if (version?.hasPrebuiltVariants) {
+				const rawVersionForVariant = version.rawVersions[newVariant];
+				if (rawVersionForVariant != null) {
+					newState.version = rawVersionForVariant;
+				}
+			}
```

Comment:
Please double check that we do not have a useEffect or something somewhere that is triggered by `developmentMode` changing that already does this so that we do not duplicate this

Reply:
We did't have any, and that's why this issue is happening at the first place :)